### ISSUE TITLE
Pi Pico W will not connect to WiFi (and bootloops) without Static IP

### ISSUE THREAD

**Issue by LewisSpring** [esphome/issues#4110]
### The problem

If I compile a Pico W firmware (eg, the default ESPHome yaml) without a static IP, the Pico gets into a boot loop where it is unable to connect to WiFi and also does not seem to initilise anything else, as the log does not mention components like API, MDNS or Logger.

### Which version of ESPHome has the issue?

2022.12.8

### What type of installation are you using?

Docker

### What platform are you using?

RP2040

### Board

Pi Pico W

### Component causing the issue

network

### Example YAML snippet

```yaml
{default esphome yaml for pico W}
```

### Anything in the logs that might be useful for us?

![image](https://user-images.githubusercontent.com/40576482/216350552-c649bbed-5c75-44bd-910f-da7769ff634e.png)
When the issue is present, terminal looks like this. Each time the Pico reboots, the terminal prints the same thing.

![image](https://user-images.githubusercontent.com/40576482/216350978-830cc735-b144-44b1-a5d4-0480d1a9a795.png)
Adding a static IP makes it boot correctly

### Additional information

Discord thread
https://discord.com/channels/429907082951524364/1069263380294152243

**Comment by DezzyTee** [comment]
Same issue

**Comment by KurdFromKurdistan** [comment]
Did you find a solution to this problem? I cannot add my RP2040 to home assistant devices or read logs wirelessly, but I can update the code wirelessly.

**Comment by 24fdasdfjsa** [comment]
> > Did you find a solution to this problem? I cannot add my RP2040 to home assistant devices or read logs wirelessly, but I can update the code wirelessly.
> 
> I haven’t played with the Pico in the last few months - by the sounds of it it’s not fixed?
> 
> I had no issues with these functions while using the static IP. Maybe try adding `use_address` in the networking configuration?

Yes it worked thank you so much for responding! It is probably an issue with how the name gets translated to the hostname... Everything works now

**Comment by bdraco** [comment]
Closing — this was reported against ESPHome 2022.12.8 and a lot has changed since then. ESPHome 2026.3.0 upgraded to arduino-pico 5.5.1 which includes numerous WiFi fixes. If anyone is still seeing bootloops without a static IP on a current version, please open a new issue.
